File: src/main/java/com/example/board/service/CommentService.java
package com.example.board.service;
import java.sql.Connection;
import java.sql.Statement;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import javax.sql.DataSource;
import org.springframework.stereotype.Service;

/**
 * 댓글 서비스 (수정됨).
 */
@Service
public class CommentService {

    private final DataSource dataSource;

    public CommentService(DataSource dataSource) {
        this.dataSource = dataSource;
    }

    public void loadComments(String postId) {
        String sql = "SELECT * FROM comments WHERE post_id = ?";
        try (Connection conn = dataSource.getConnection();
             PreparedStatement pstmt = conn.prepareStatement(sql)) {
            pstmt.setString(1, postId);
            ResultSet rs = pstmt.executeQuery();

            while (rs.next()) {
                String body = rs.getString("body");
                System.out.println(body);
            }
        } catch (Exception e) {
            // 안전한 로그 처리나 사용자 친화적 에러 처리를 대신 적용.
            e.printStackTrace();
        }
    }
}

File: app/services/comment_service.py
from contextlib import closing

class CommentService:
    """
    Comment service (refactored from Java to Python).
    """

    def __init__(self, data_source):
        self.data_source = data_source

    def load_comments(self, post_id):
        sql = "SELECT * FROM comments WHERE post_id = %s"
        try:
            with closing(self.data_source.get_connection()) as conn:
                with closing(conn.cursor()) as cursor:
                    cursor.execute(sql, (post_id,))
                    for row in cursor.fetchall():
                        print(row["body"])
        except Exception as e:
            # Safe logging or user-friendly error handling.
            print(f"An error occurred: {str(e)}")